server/src/main/java/com/cloud/network/vpc/VpcManagerImpl.java

```diff
@@ -2109,6 +2151,13 @@ public boolean cleanupVpcResources(final long vpcId, final Account caller, final
 
         VpcVO vpc = vpcDao.findById(vpcId);
         annotationDao.removeByEntityType(AnnotationService.EntityType.VPC.name(), vpc.getUuid());
+
+        ASNumberVO asNumber = asNumberDao.findByZoneAndVpcId(vpc.getZoneId(), vpc.getId());
```

Comment:
this can be moved into the method `bgpService.releaseASNumber`

Reply:
@weizhouapache I prefer to keep it as it is so releaseASNumber can accept the AS number to be released